include/secp256k1_bulletproofs.h

```diff
@@ -0,0 +1,182 @@
+
+/** Verifies a single bulletproof (aggregate) rangeproof
+ *  Returns: 1: rangeproof was valid
+ *           0: rangeproof was invalid, or out of memory
+ *  Args:       ctx: pointer to a context object initialized for verification (cannot be NULL)
+ *          scratch: scratch space with enough memory for verification (cannot be NULL)
+ *             gens: generator set with at least 2*nbits*n_commits many generators (cannot be NULL)
+ *  In:       proof: byte-serialized rangeproof (cannot be NULL)
+ *             plen: length of the proof
+ *        min_value: array of minimum values to prove ranges above, or NULL for all-zeroes
+ *           commit: array of pedersen commitment that this rangeproof is over (cannot be NULL)
+ *        n_commits: number of commitments in the above array (cannot be 0)
+ *            nbits: number of bits proven for each range
+ *        value_gen: generator multiplied by value in pedersen commitments (cannot be NULL)
+ *     extra_commit: additonal data committed to by the rangeproof
+ * extra_commit_len: length of additional data
+ */
+SECP256K1_WARN_UNUSED_RESULT SECP256K1_API int secp256k1_bulletproof_rangeproof_verify(
+    const secp256k1_context* ctx,
+    secp256k1_scratch_space* scratch,
+    const secp256k1_bulletproof_generators *gens,
+    const unsigned char* proof,
+    size_t plen,
+    const uint64_t* min_value,
+    const secp256k1_pedersen_commitment* commit,
+    size_t n_commits,
+    size_t nbits,
+    const secp256k1_generator* value_gen,
+    const unsigned char* extra_commit,
+    size_t extra_commit_len
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(7) SECP256K1_ARG_NONNULL(10);
+
+/** Batch-verifies multiple bulletproof (aggregate) rangeproofs of the same size using same generator
+ *  Returns: 1: all rangeproofs were valid
+ *           0: some rangeproof was invalid, or out of memory
+ *  Args:       ctx: pointer to a context object initialized for verification (cannot be NULL)
+ *          scratch: scratch space with enough memory for verification (cannot be NULL)
+ *             gens: generator set with at least 2*nbits*n_commits many generators (cannot be NULL)
+ *  In:       proof: array of byte-serialized rangeproofs (cannot be NULL)
+ *         n_proofs: number of proofs in the above array, and number of arrays in the `commit` array
+ *             plen: length of every individual proof
+ *        min_value: array of arrays of minimum values to prove ranges above, or NULL for all-zeroes
+ *           commit: array of arrays of pedersen commitment that the rangeproofs is over (cannot be NULL)
+ *        n_commits: number of commitments in each element of the above array (cannot be 0)
+ *            nbits: number of bits in each proof
+ *        value_gen: generator multiplied by value in pedersen commitments (cannot be NULL)
+ *     extra_commit: array of additonal data committed to by the rangeproof
```

Comment:
(individual fields can only be NULL if the corresponding extra_commit_len` equals 0)